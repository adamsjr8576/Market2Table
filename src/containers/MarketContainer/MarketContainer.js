import React from 'react';
import { connect } from 'react-redux';
import Market from '../../Components/Market/Market.js';

const MarketContainer = ({ markets }) => {
  const marketsList = markets.map(market => {
    return (
      <Market
        key={market.id}
        id={market.id}
        marketName={market.marketname}
      />
    )
  })
  return (
    <main className="market-container-main">
      <section className="market-list-section">
        {marketsList}
      </section>
      <section className="markets-map-section">
        <p>MAPPPPP</p>
      </section>
    </main>
  )
}

export const mapStateToProps = state => ({
  markets: state.farmersMarkets,
  zipCode: state.zipCode
});

export default connect(mapStateToProps)(MarketContainer);
