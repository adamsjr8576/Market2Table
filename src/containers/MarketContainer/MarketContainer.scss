.market-container-main {
  display: flex;
  align-items: center;
  justify-content: space-around;
  height: 83vh;
}

.market-list-section {
  height: 500px;
  display: flex;
  flex-direction: column;
  align-items: center;
  overflow: auto;
  width: 600px;
  border-top: 4px solid #FFE7BF;
  border-bottom: 4px solid #FFE7BF;
}

.markets-map-section {
  height: 500px;
  width: 700px;
  text-align: center;
}

.favorites-message {
  padding: 20px;
  text-align: center;
  background-image: linear-gradient(rgba(0, 0, 0, .7), rgba(0, 0, 0, .7));
  color: white;
  border: 3px solid white;
  border-radius: 30px;
}
