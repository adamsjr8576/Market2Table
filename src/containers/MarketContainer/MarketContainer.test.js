import React from 'react';
import {shallow} from 'enzyme';
import { MarketContainer, mapStateToProps } from './MarketContainer';

describe('MarketContainer', () => {
  describe('MarketContainer Component', () => {
    let mockMarkets;
    let mockZipCode;
    let mockFavorites;
    let mockPath;

    beforeEach(() => {
      mockMarkets =
        [
           {
             id: "1002192",
             marketname: "Ballston FRESHFARM Market",
             favorite: false,
             GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
             Address: "901 N Taylor St, Ballston, Virginia, 22203",
             Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
             Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
             latitude: "38.881112",
             longitude: "-77.112179"
           },
           {
             id: "22008839",
             marketname: "Ballston Farmers Market",
             favorite: false,
             GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
             Address: "901 N Taylor St, Ballston, Virginia, 22203",
             Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
             Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
             latitude: "38.881112",
             longitude: "-77.112179"
           }
        ]
    });

    it('should match the snapshot with path on favorites containing a favorite', () => {
      mockFavorites = [
        {
          id: "22008839",
          marketname: "Ballston Farmers Market",
          favorite: true,
          GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
          Address: "901 N Taylor St, Ballston, Virginia, 22203",
          Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
          Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
          latitude: "38.881112",
          longitude: "-77.112179"
        }
      ]
      mockPath = '/markets/favorites';
      mockZipCode = '80401';
      const wrapper = shallow(
        <MarketContainer
          markets={mockMarkets}
          zipCode={mockZipCode}
          favorites={mockFavorites}
          path={mockPath}
        />)

      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot with path on favorites containing no favorites', () => {
      mockFavorites = []
      mockPath = '/markets/favorites';
      mockZipCode = '80401';
      const wrapper = shallow(
        <MarketContainer
          markets={mockMarkets}
          zipCode={mockZipCode}
          favorites={mockFavorites}
          path={mockPath}
        />)

      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot with path on markets', () => {
      mockFavorites = []
      mockPath = '/markets';
      mockZipCode = '80401';
      const wrapper = shallow(
        <MarketContainer
          markets={mockMarkets}
          zipCode={mockZipCode}
          favorites={mockFavorites}
          path={mockPath}
        />)

      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot with path on markets and when there is no zip code', () => {
      mockFavorites = []
      mockPath = '/markets';
      mockZipCode = '';
      const wrapper = shallow(
        <MarketContainer
          markets={mockMarkets}
          zipCode={mockZipCode}
          favorites={mockFavorites}
          path={mockPath}
        />)

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with markets, zipCode, favorites', () => {
      const mockMarkets = [
         {
           id: "1002192",
           marketname: "Ballston FRESHFARM Market",
           favorite: false,
           GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
           Address: "901 N Taylor St, Ballston, Virginia, 22203",
           Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
           Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
           latitude: "38.881112",
           longitude: "-77.112179"
         },
         {
           id: "22008839",
           marketname: "Ballston Farmers Market",
           favorite: false,
           GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
           Address: "901 N Taylor St, Ballston, Virginia, 22203",
           Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
           Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
           latitude: "38.881112",
           longitude: "-77.112179"
         }];
      const mockState = {
        farmersMarkets: mockMarkets,
        zipCode: '80401',
        favorites: []
      }

      const expected = {
        markets: mockMarkets,
        zipCode: '80401',
        favorites: []
      };

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});
