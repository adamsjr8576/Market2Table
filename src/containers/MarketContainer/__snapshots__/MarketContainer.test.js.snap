// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketContainer MarketContainer Component should match the snapshot with path on favorites containing a favorite 1`] = `
<main
  className="market-container-main"
>
  <section
    className="market-list-section"
  >
    <Connect(Market)
      image="redChili.png"
      key="22008839"
      market={
        Object {
          "Address": "901 N Taylor St, Ballston, Virginia, 22203",
          "GoogleLink": "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
          "Products": "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
          "Schedule": "June - October Thursday 3:00 PM to 7:00 PM",
          "favorite": true,
          "id": "22008839",
          "latitude": "38.881112",
          "longitude": "-77.112179",
          "marketname": "Ballston Farmers Market",
        }
      }
    />
  </section>
  <section
    className="markets-map-section"
  >
    <p>
      MAPPPPP
    </p>
  </section>
</main>
`;

exports[`MarketContainer MarketContainer Component should match the snapshot with path on favorites containing no favorites 1`] = `
<main
  className="market-container-main"
>
  <section
    className="market-list-section"
  >
    <h1
      className="favorites-message"
    >
      You have no favorites selected!
    </h1>
  </section>
  <section
    className="markets-map-section"
  >
    <p>
      MAPPPPP
    </p>
  </section>
</main>
`;

exports[`MarketContainer MarketContainer Component should match the snapshot with path on markets 1`] = `
<main
  className="market-container-main"
>
  <section
    className="market-list-section"
  >
    <Connect(Market)
      image="chilli.png"
      key="1002192"
      market={
        Object {
          "Address": "901 N Taylor St, Ballston, Virginia, 22203",
          "GoogleLink": "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
          "Products": "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
          "Schedule": "June - October Thursday 3:00 PM to 7:00 PM",
          "favorite": false,
          "id": "1002192",
          "latitude": "38.881112",
          "longitude": "-77.112179",
          "marketname": "Ballston FRESHFARM Market",
        }
      }
    />
    <Connect(Market)
      image="chilli.png"
      key="22008839"
      market={
        Object {
          "Address": "901 N Taylor St, Ballston, Virginia, 22203",
          "GoogleLink": "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
          "Products": "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
          "Schedule": "June - October Thursday 3:00 PM to 7:00 PM",
          "favorite": false,
          "id": "22008839",
          "latitude": "38.881112",
          "longitude": "-77.112179",
          "marketname": "Ballston Farmers Market",
        }
      }
    />
  </section>
  <section
    className="markets-map-section"
  >
    <p>
      MAPPPPP
    </p>
  </section>
</main>
`;

exports[`MarketContainer MarketContainer Component should match the snapshot with path on markets and when there is no zip code 1`] = `
<main
  className="market-container-main"
>
  <Connect(ZipCodeForm)
    path="/markets"
  />
  <section
    className="market-list-section"
  >
    <Connect(Market)
      image="chilli.png"
      key="1002192"
      market={
        Object {
          "Address": "901 N Taylor St, Ballston, Virginia, 22203",
          "GoogleLink": "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
          "Products": "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
          "Schedule": "June - October Thursday 3:00 PM to 7:00 PM",
          "favorite": false,
          "id": "1002192",
          "latitude": "38.881112",
          "longitude": "-77.112179",
          "marketname": "Ballston FRESHFARM Market",
        }
      }
    />
    <Connect(Market)
      image="chilli.png"
      key="22008839"
      market={
        Object {
          "Address": "901 N Taylor St, Ballston, Virginia, 22203",
          "GoogleLink": "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
          "Products": "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
          "Schedule": "June - October Thursday 3:00 PM to 7:00 PM",
          "favorite": false,
          "id": "22008839",
          "latitude": "38.881112",
          "longitude": "-77.112179",
          "marketname": "Ballston Farmers Market",
        }
      }
    />
  </section>
  <section
    className="markets-map-section"
  >
    <p>
      MAPPPPP
    </p>
  </section>
</main>
`;
